pkg/gitreceive/build.go

```diff
@@ -112,6 +110,20 @@ func build(
 		}
 	}
 
+	_, disableCaching := appConf.Values["DEIS_DISABLE_CACHE"]
+	slugBuilderInfo := NewSlugBuilderInfo(appName, gitSha.Short(), disableCaching)
+
+	if slugBuilderInfo.DisableCaching() {
+		log.Debug("caching disabled for app %s", appName)
+		// If cache file exists, delete it
+		if _, err := storageDriver.Stat(context.Background(), slugBuilderInfo.CacheKey()); err == nil {
```

Comment:
I was incorrect, deleting a object on S3 is free. That being said, I suspect it's better practice to check if the file exists *before* trying to delete it, but that's me.